tests/Types/VarCharTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Types;
+
+use Doctrine\DBAL\Driver\Encodings;
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\VarCharType;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class VarCharTest extends TestCase
+{
+    /** @var AbstractPlatform|MockObject */
+    private $platform;
+
+    /** @var VarCharType */
+    private $type;
+
+    protected function setUp(): void
+    {
+        $this->platform = $this->createMock(AbstractPlatform::class);
+        $this->type     = new VarCharType();
+    }
+
+    public function testReturnsBindingType(): void
+    {
+        self::assertEquals($this->type->getBindingType(), ParameterType::STRING | Encodings::ASCII);
+    }
+
+    /**
+     * @param array<string, int|bool> $column
+     *
+     * @dataProvider sqlDeclarationDataProvider
+     */
+    public function testReturnsSQLDeclaration(string $expectedSql, array $column): void
```

Comment:
Does it even need to be tested? The sole purpose of this type is to override the default way of binding variables to the statement. The SQL declaration should be identical to the String type. Isn't it?